crates/extrinsics/src/instantiate.rs

```diff
@@ -317,48 +375,68 @@ impl Exec {
             .ok_or_else(|| anyhow!("Failed to find Instantiated event"))?;
 
         let token_metadata = TokenMetadata::query(&self.client).await?;
-        self.display_result(&result, None, instantiated.contract, &token_metadata)
-            .await
+        Ok(InstantiateExecResult {
+            result,
+            code_hash: None,
+            contract_address: instantiated.contract,
+            token_metadata,
+        })
     }
 
-    async fn display_result(
+    pub async fn instantiate(
         &self,
-        result: &ExtrinsicEvents<DefaultConfig>,
-        code_hash: Option<CodeHash>,
-        contract_address: subxt::utils::AccountId32,
-        token_metadata: &TokenMetadata,
+        gas_limit: Weight,
```

Comment:
Cherry on top would be to have this as an `Option`, where this method would first issue a dry-run to estimate the gas for you. Same for the `call` version.